### PR TITLE
Moved auth information at the instance level

### LINKED ISSUE #8: Accept auth parameters when instanciating Smooch
By doing this, we'll reduce the amount of work needed to talk with the API. It should accept either a generated jwt, an app token, or a keyId-secret-scope combination for the jwt.

We should put a check on the browser wrapper to throw and error if the jwt combination is used. It would throw anyway since the jwt utils won't be there.

### PR DESCRIPTION
**Pull request by lemieux**
This simplifies the auth handling and fixes #8 .

@alavers

### CODE REVIEW

File: src/api/webhooks.js [hunk 1]
```diff
@@ -44,9 +44,9 @@ export class WebhooksApi extends BaseApi {
    * @param  {AuthCredentials} auth
    * @return {APIResponse}
    */
-  list(auth) {
+  list() {
```
Comment (alavers) [inline src/api/webhooks.js]: Perhaps this belongs as  a TODO, but `validateAuth` could also be moved to base. Specific instances of BaseApi should then simply specify an array of allowed auth mechanisms eg `['jwt', 'appToken']`

File: tests/specs/api/appUsers.spec.js [hunk 2]
```diff
@@ -1,23 +1,21 @@
...
     appToken: 'token'
-  };
+  });
+
   let httpSpy;
   let api;
 
   beforeEach(() => {
     httpSpy = httpMock.mock();
     let AppUsersApi = require('../../../src/api/appUsers').AppUsersApi;
```
Comment (alavers) [inline tests/specs/api/appUsers.spec.js]: is there an advantage to putting this require inline?

**Comment by alavers** [comment]
Rest looks muy bueno :+1: :cake: :facepunch:

File: tests/specs/api/appUsers.spec.js (hunk 2 above)
Comment (lemieux) [inline tests/specs/api/appUsers.spec.js]: I'll check if it change anything but I had the impression I would need to mock the http part before requiring the class... to be tested.

**Comment by lemieux** [comment]
Babel is being difficult for some reason and make the tests fails.. I'll investigate that and perhaps fix the dependencies version in package.json

File: src/api/webhooks.js (hunk 1 above)
Comment (lemieux) [inline src/api/webhooks.js]: hmm good idea I like that

Comment (lemieux) [inline src/api/webhooks.js]: @alavers I changed all apis to do that... I hate that I need to do it in all methods... in a ES7 world I'd use decorators instead... but we're not there yet and they are still just a proposal.

**Comment by lemieux** [comment]
I still need to update the readme though.. too lazy to do that tonight

**Pull request merged**